lib/commands/init.js

```diff
@@ -166,6 +167,9 @@ class Init extends BaseCommand {
 
     try {
       const data = await input.read(() => initJson(path, initFile, this.npm.config))
+      if (this.npm.config.isDefault('init-private')) {
+        delete data.private
```

Comment:
Ah yes, this is one of the pitfalls of doing this kind of immediate return inside the try/catch block.  You'll either have to refactor or wrap this in an `if (data)` clause.

Reply:
@wraithgar what do you think about doing 

```javascript
const initPrivate = getConfig('private')
if (initPrivate === true) {
  exports.private = true
}
```

In the init-package-json instead of doing something 'smart' here? That is looking way cleaner and easier to me

Reply:
That makes sense.  We don't need to add it to the package.json unless it's true, basically.